downloadDirectory/src/main/java/org/reactome/release/downloadDirectory/ProtegeExporter.java

```diff
@@ -0,0 +1,404 @@
+		// Build the process.
+		ProcessBuilder processBuilder = new ProcessBuilder();
+		processBuilder.command(cmdArgs)
+					.directory(Paths.get(this.pathToWrapperScript).toFile())
+					.inheritIO();
+		logger.info("Command is: `{}`", String.join(" ", processBuilder.command()));
+		return processBuilder;
+	}
+
+	/**
+	 * TARs the protege files.
+	 * @param pathToProtegeFiles The path to where the protege archive files are. They need to be collected into a single archive.
+	 */
+	private void tarProtegeFiles()
+	{
+		try(FileOutputStream protegeTar = new FileOutputStream(PROTEGE_ARCHIVE_PATH);
+			TarArchiveOutputStream tarOutStream = new TarArchiveOutputStream(protegeTar);)
+		{
+			Files.list(Paths.get(PROTEGE_FILES_DIR)).forEach( protegeFile -> 
+			{
+				// Sanity check on the file: Since we're just iterating through a directory, there could be other stuff in there.
+				// Only continue if:
+				// 	1) the file is a NORMAL file (not a directory) and
+				// 	2) it matches the pattern Reactome_pathway_.*\.tar\.gz
+				if (protegeFile.toFile().isFile() && protegeFile.getFileName().toString().matches("Reactome_pathway_.*\\.tar\\.gz"))
+				{
+					logger.info("Adding {} to protege_files.tar", protegeFile.toString());
+					addFileToTar(tarOutStream, protegeFile);
+				}
+			});
+			tarOutStream.finish();
+		}
+		catch (IOException e)
+		{
+			e.printStackTrace();
+		}
+		
+		// Now that we're finished creating the tar file, move it to the download directory. Overwrite existing.
+		try
+		{
+			Files.move(Paths.get(PROTEGE_ARCHIVE_PATH), Paths.get(this.downloadDirectory + "/protege_files.tar"), StandardCopyOption.REPLACE_EXISTING);
+		}
+		catch (IOException e)
+		{
+			logger.error("An error occurred while trying to move {} to the download directory ({}). You may need to move it manually.", PROTEGE_ARCHIVE_PATH, this.downloadDirectory);
+			e.printStackTrace();
+		}
+	}
+
+	/**
+	 * Adds a file to a tar file.
+	 * @param tarOutStream - the output stream for the tar file that you want to add to.
+	 * @param protegeFile - the Path to the input file.
+	 * @throws IOException
+	 */
+	private static void addFileToTar(TarArchiveOutputStream tarOutStream, Path protegeFile)
+	{
+		try(InputStream is = new FileInputStream(protegeFile.toFile());)
+		{
+			// Create an entry with the name being the name of the file.
+			TarArchiveEntry entry = new TarArchiveEntry(protegeFile.getFileName().toString());
+			entry.setSize(protegeFile.toFile().length());
+			tarOutStream.putArchiveEntry(entry);
+			
+			final int len = 1024;
+			byte[] buff = new byte[len];
+			long bytesRead = is.read(buff, 0, len);
+			while (bytesRead > 0)
+			{
+				// If bytesRead + len > entrySize then an exception will be thrown, so always take min of len,bytesRead. 
+				tarOutStream.write(buff, 0, (int) Math.min(len, bytesRead));
+				bytesRead = is.read(buff, 0, len);
```

Comment:
Could this be done in the while condition itself so it is not repeated from line 313:
```
long bytesRead;
while ((bytesRead = is.read(buff, 0, len)) > 0) {
    tarOutStream.write(buff, 0, (int) Math.min(len, bytesRead));
}

Reply:
This might be valid Java, but it's not a _normal_ pattern, and I would call it a code smell. I find it weird, unexpected, and confusion.

Reply:
Yeah, the while expression is a bit verbose.  If there's a way to consolidate the calls to `is.read(buff,0,len)` so it happens once in the code instead of twice, though, that'd be good.